### ISSUE TITLE
Spectrum coloring based on b-factors

### ISSUE THREAD

**Issue by josemduarte** [dkoes/3Dmol.js#88]
If I understand it, the current spectrum coloring will color things based on residue or atom indices. For instance I can see that applying spectrum on a chain, I get an n-terminal to c-terminal spectrum.

It would be really cool to have the pymol-like feature, where spectrum can use the b-factors for the coloring. In fact the [spectrum command in pymol](http://www.pymolwiki.org/index.php/Spectrum) can do quite a lot of things, all of those could be really useful.

Another very nice feature would be that spectrum worked also in surface representation (it looks like that's already in issue #79 ). Spectrum on b-factors (temperature factors or whatever property is actually written in the b-factors column) is especially useful in surface representation.

**Comment by dkoes** [comment]
I've been working on improving this.  The old way to color by properties is this:
```
glviewer.mapAtomProperties(function(a) { a.properties['b'] = a.b;});
m.setColorByProperty({}, 'b', new $3Dmol.Gradient.Sinebow(0,50));
```
This is a little cumbersome since you have to copy the property value and we're making the default color of each atom correspond to the property mapped color.  Really, the color should be a property of the style, not the atom. 

I've committed some code (13f3fc0) that let's you define property mappings as part of the style's color scheme. I've also made it so that b-factors can be referenced as properties.
```
glviewer.setStyle({},{sphere:{colorscheme:{prop:'b',gradient: new $3Dmol.Gradient.Sinebow(0,50)}}});
```
There's also a function for computing the range of values of a property if you don't want to hard-code the gradient:
```
var gradient = new $3Dmol.Gradient.Sinebow($3Dmol.getPropertyRange(glviewer.selectedAtoms(),'b')); 
```

This should work for surfaces as well.
```
viewer.addSurface($3Dmol.SurfaceType.VDW, {colorscheme:{prop:'b',gradient: gradient}});
```

You can colormap any numerical property.  For example, mol2 and pqr files have a partialCharge property.

The 'spectrum' colorscheme will eventually be just shorthand for specifying a mapping colorscheme from the residue ids, as opposed to the special purpose code it is now (this is the fix to issue #79).

**Comment by josemduarte** [comment]
That's awesome, thanks a lot for the explanation. I'll give it a try as soon as I can.
